src/libpcp_web/src/query_parser.y

```diff
@@ -524,6 +526,50 @@ func_sid
 		  lp->yy_np->left = $3;
 		  $$ = lp->yy_series.expr = lp->yy_np;
 		}
+	| L_NTH_PERCENTILE_INST L_LPAREN sid_vec L_COMMA L_INTEGER L_RPAREN
+		{ lp->yy_np = newnode(N_NTH_PERCENTILE_INST);
+		  lp->yy_np->left = $3;
+		  if (atoi($5) < 0 || atoi($5) > 100){
```

Comment:
Instead of atoi, let's use strtoul here - that way, we can do some error checking.
Also, more efficient if we call it just once and save the result in a temporary variable rather than doing the string conversion twice.

Reply:
It gets updated.